### DISCUSSION TITLE
Varied sizes of image crops

### DISCUSSION THREAD

**Discussion by ChrisB-TL** [area17/twill#1556] [Q&A] [answered]
Hello

I just wanted to check whether there is an "out of the box" way to obtain a link for different versions of a single crop?

E.g., lets say I have "cover" role, which has a square crop. The square crop is 1000 x 1000, which is too big to load on a listing page. I'd want a 100 x 100 version of that same crop.

Obviously I could do this manually by loading the mediable, doing some calculations, and then passing the relevant params to one of the ImageService methods, but I just wanted to check I wouldn't be inventing work for myself....! And if I do do that, am I right in thinking getUrlWithFocalCrop would be the method to use?

Many thanks

Chris

**Comment by pboivin** [comment, marked as answer]
Hi @ChrisB-BTL,

If you're using the Glide service (or any service that supports dynamic transformations), you can pass an array of parameters as a third argument to `image()` :
```
$page->image('cover', 'square', ['w' => 100]);
```

You can find all the Glide transformations here : https://glide.thephpleague.com/2.0/api/quick-reference/

I believe Imgix is using a very similar set.

**Reply by ChrisB-TL** [reply]
That's great, I thought they'd be a simple way! Thank you.
